main.go

```diff
@@ -167,38 +168,75 @@ func updateNodesStats(config *Config) {
 	}
 }
 
-func runRequestLoop(node Node) {
-	for {
-		getOverview(node.Url, node.Uname, node.Password)
-		getNumberOfMessages(node.Url, node.Uname, node.Password)
+func requestData(node Node) {
+	defer func() {
+		if r := recover(); r != nil {
+			dt := 10 * time.Second
+			time.Sleep(dt)
+		}
+	}()
 
-		log.Info("Metrics updated successfully.")
+	getOverview(node.Url, node.Uname, node.Password)
+	getNumberOfMessages(node.Url, node.Uname, node.Password)
 
-		dt, err := time.ParseDuration(node.Interval)
-		if err != nil {
-			log.Warn(err)
-			dt = 30 * time.Second
-		}
-		time.Sleep(dt)
+	log.Info("Metrics updated successfully.")
+
+	dt, err := time.ParseDuration(node.Interval)
+	if err != nil {
+		log.Warn(err)
+		dt = 30 * time.Second
 	}
+	time.Sleep(dt)
 }
 
-func newConfig(path string) (*Config, error) {
-	var config Config
+func runRequestLoop(node Node) {
+	for {
+		requestData(node)
+	}
+}
+
+func loadConfig(path string, c *Config) bool {
+	defer func() {
+		if r := recover(); r != nil {
+			dt := 10 * time.Second
+			time.Sleep(dt)
+		}
+	}()
 
 	file, err := ioutil.ReadFile(path)
 	if err != nil {
 		log.Error(err)
-		return nil, err
+		panic(err)
+	}
+
+	err = json.Unmarshal(file, c)
+	if err != nil {
+		log.Error(err)
+		panic(err)
+	}
+	return true
+}
+
+func runLoadConfigLoop(path string, c *Config) {
```

Comment:
Makes sense, that's what I thought but saw that config was only loaded once for now.